internal/storage/payload_writer.go

```diff
@@ -168,16 +183,20 @@ func (w *NativePayloadWriter) AddBoolToPayload(data []bool) error {
 		return errors.New("can't add empty msgs into payload")
 	}
 
+	if len(data) != len(nullBitset) {
```

Comment:
add condition `nullBitset != nil`, then we don't need to construct a bitset if the field is not nullable, allows to pass nil nullBitset for this case